src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -162,6 +162,12 @@ internal void TraceCompilationDiagnostics(ImmutableArray<Diagnostic> diagnostics
             foreach (var diagnostic in diagnostics.Where(d => !d.IsSuppressed))
             {
                 FunctionLogger.Log(diagnostic.Severity.ToLogLevel(), 0, logState, null, (s, e) => diagnostic.ToString());
+                // Check if script compilation failed due to missing assembly (CS0246)
+                if (diagnostic.Id.Equals(DotNetConstants.TypeOrNamespaceNotFoundCompilerErrorCode))
```

Comment:
This logic should be in the `DotNetFunctionInvoker`.

There's some existing logic that inspects compilation diagnostics and adds to the list of diagnostics. (look for `GetFunctionDiagnostics`)